src/rs.cpp

```diff
@@ -1749,3 +1787,51 @@ void rs2_log(rs2_log_severity severity, const char * message, rs2_error ** error
     }
 }
 HANDLE_EXCEPTIONS_AND_RETURN(, severity, message)
+
+void rs2_loopback_enable(const rs2_device* device, const char* from_file, rs2_error** error) try
+{
+    VALIDATE_NOT_NULL(device);
+    VALIDATE_NOT_NULL(from_file);
+
+    auto loopback = VALIDATE_INTERFACE(device->device, librealsense::tm2_extensions);
+    loopback->enable_loopback(from_file);
+
+}
+HANDLE_EXCEPTIONS_AND_RETURN(, device, from_file)
+
+void rs2_loopback_disable(const rs2_device* device, rs2_error** error) try
+{
+    VALIDATE_NOT_NULL(device);
+
+    auto loopback = VALIDATE_INTERFACE(device->device, librealsense::tm2_extensions);
+    loopback->disable_loopback();
+}
+HANDLE_EXCEPTIONS_AND_RETURN(, device)
+
+int rs2_loopback_is_enabled(const rs2_device* device, rs2_error** error) try
+{
+    VALIDATE_NOT_NULL(device);
+
+    auto loopback = VALIDATE_INTERFACE(device->device, librealsense::tm2_extensions);
+    return loopback->is_enabled() ? 1 : 0;
+}
+HANDLE_EXCEPTIONS_AND_RETURN(0, device)
+
+void rs2_connect_tm2_controller(const rs2_device* device, const unsigned char* mac, rs2_error** error) try
+{
+    VALIDATE_NOT_NULL(device);
+    VALIDATE_NOT_NULL(mac);
+
+    auto tm2 = VALIDATE_INTERFACE(device->device, librealsense::tm2_extensions);
+    tm2->connect_controller({ mac[0], mac[1], mac[2], mac[3], mac[4], mac[5] });
```

Comment:
Refactor/Add array boundaries check

Reply:
the array size should be exactly 6, how do you do that with C?